README.md

```diff
@@ -28,57 +32,68 @@ tarpc = "0.5.0"
-    assert_eq!("Hello, Mom!", client.hello("Mom".into()).unwrap());
-    drop(client);
-    server_handle.shutdown();
+    let _server = HelloServer.listen(addr).unwrap();
+    let client = SyncClient::connect(addr).unwrap();
+    assert_eq!("Hello, Mom!", client.hello(&"Mom".to_string()).unwrap());
 }
 ```
 
-The `service!` macro expands to a collection of items that collectively form an rpc service. In the
-above example, the macro is called within the `hello_service` module. This module will contain a
-`Client` (and `AsyncClient`) type, and a `Service` trait. The trait provides default `fn`s for
-starting the service: `spawn` and `spawn_with_config`, which start the service listening over an
-arbitrary transport. A `Client` (or `AsyncClient`) can connect to such a service. These generated
-types make it easy and ergonomic to write servers without dealing with sockets or serialization
-directly. See the tarpc_examples package for more sophisticated examples.
+The `service!` macro expands to a collection of items that collectively form an
+rpc service. In the above example, the macro is called within the
+`hello_service` module. This module will contain `SyncClient`, `AsyncClient`,
+and `FutureClient` types, and `SyncService` and `AsyncService` traits.  There is
+also a `ServiceExt` trait that provides starter `fn`s for services, with an
+umbrella impl for all services.  These generated types make it easy and
+ergonomic to write servers without dealing with sockets or serialization
+directly. Simply implement one of the generated traits, and you're off to the
+races! See the tarpc_examples package for more examples.
 
 ## Documentation
 Use `cargo doc` as you normally would to see the documentation created for all
 items expanded by a `service!` invocation.
 
 ## Additional Features
-- Connect over any transport that `impl`s the `Transport` trait.
+- Configurable server rate limiting.
+- Automatic client retries with exponential backoff when server is busy.
 - Concurrent requests from a single client.
-- Any type that `impl`s `serde`'s `Serialize` and `Deserialize` can be used in the rpc signatures.
-- Attributes can be specified on rpc methods. These will be included on both the `Service` trait
-  methods as well as on the `Client`'s stub methods.
+- Backed by an mio `EventLoop`, protecting services (including `SyncService`s)
+  from slowloris attacks.
+- Run any number of clients on a single client event loop.
+- Run any number of services on a single service event loop.
+- Configure clients and services to run on a custom event loop, defaulting to
+  the global event loop.
+- Any type that `impl`s `serde`'s `Serialize` and `Deserialize` can be used in
+  the rpc signatures.
+- Attributes can be specified on rpc methods. These will be included on both the
+  services' trait methods as well as on the clients' stub methods.
 - Just like regular fns, the return type can be left off when it's `-> ()`.
```

Comment:
I guess I'd lean towards not listing these two, since intuitively I would expect the macro to handle most of the same syntax as rust fns.